test/extended/operators/tolerations.go

```diff
@@ -0,0 +1,77 @@
+package operators
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = Describe("[Feature:Platform][Smoke] Managed cluster", func() {
+	oc := exutil.NewCLIWithoutNamespace("operators")
+
+	It("should ensure pods do not tolerate all taints unless on whitelist", func() {
+		// iterate over the references to find valid images
+		pods, err := oc.KubeFramework().ClientSet.CoreV1().Pods("").List(metav1.ListOptions{})
+		if err != nil {
+			e2e.Failf("unable to list pods: %v", err)
+		}
+
+		// list of pods that use images not in the release payload
+		invalidPodTolerations := sets.NewString()
+		// a pod in a namespace that begins with kube-* or openshift-* must come from our release payload
+		// TODO components in openshift-operators may not come from our payload, may want to weaken restriction
+		namespacePrefixes := sets.NewString("kube-", "openshift-")
+		excludedNamespaces := sets.NewString("openshift-kube-apiserver", "openshift-kube-controller-manager", "openshift-kube-scheduler", "openshift-etcd")
+		for _, pod := range pods.Items {
+			// exclude non-control plane namespaces
+			if !hasPrefixSet(pod.Namespace, namespacePrefixes) {
+				continue
+			}
+			// exclude static pod managed namespaces
+			if hasPrefixSet(pod.Namespace, excludedNamespaces) {
+				continue
+			}
+			// exclude pods started by DaemonSets
+			if ownedByDaemonSet(pod) {
+				continue
+			}
+			for _, toleration := range pod.Spec.Tolerations {
+				if toleration.Operator == v1.TolerationOpExists {
+					if toleration.Key == "" || toleration.Key == "node.kubernetes.io/unreachable" {
```

Comment:
and effect== `NoExecute`, there could be pods that may want to to be scheduled onto `unreachable` nodes too perhaps `sdn`?

Reply:
Start with the strict check and see what fails.